packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/AmplifyApiRestTest.kt

```diff
@@ -0,0 +1,429 @@
+                )
+        )
+
+    }
+
+    // Invalid response code throws error
+    @Test
+    fun test_get_invalid_input_map_error(){
+
+        var data = getFailedData
+        var restResponse = RestResponse(400, data)
+
+        Mockito.doAnswer { invocation ->
+            (invocation.arguments[1] as Consumer<RestResponse>).accept(
+                    restResponse
+            )
+            mockRestOperation
+        }.`when`(mockApi).get(
+                any(RestOptions::class.java),
+                any(),
+                any())
+
+        var arguments : Map<String, Any> = mapOf(
+        )
+
+        flutterPlugin.onMethodCall(
+                MethodCall("get", arguments),
+                mockResult
+        )
+
+        verify(mockResult).error(
+                "AmplifyException",
+                "ERROR_CASTING_INPUT_IN_PLATFORM_CODE",
+                mapOf(
+                        "PLATFORM_EXCEPTIONS" to mapOf(
+                                "platform" to "Android",
+                                "localizedErrorMessage" to "null cannot be cast to non-null type kotlin.collections.Map<kotlin.String, *>",
+                                "recoverySuggestion" to "",
+                                "errorString" to "kotlin.TypeCastException: null cannot be cast to non-null type kotlin.collections.Map<kotlin.String, *>"
+                        )
+                )
+        )
+    }
+
+    // Cancellation of ongoing rest operation succeeds
+    @Test
+    fun test_cancel_get_returns_success(){
+
+        var data = getSuccessData
+        var restResponse = RestResponse(200, data)
+        var cancelToken = "someCode"
+
+        Mockito.doAnswer { invocation ->
+            Assert.assertEquals(
+                    RestOptions.builder().addPath("/items").build(),
+                    invocation.arguments[0]
+            )
+            mockRestOperation
+        }.`when`(mockApi).get(
+                any(RestOptions::class.java),
+                any(),
+                any())
+
+        var arguments : Map<String, Any> = mapOf(
+                "restOptions" to mapOf(
+                        "path" to "/items"
+                ),
+                "cancelToken" to cancelToken
+        )
+
+        flutterPlugin.onMethodCall(
+                MethodCall("get", arguments),
+                mockResult
+        )
+
+        val mockCancelResult: MethodChannel.Result = mock(MethodChannel.Result::class.java)
+
+        flutterPlugin.onMethodCall(
+                MethodCall("cancel", cancelToken),
+                mockCancelResult
+        )
+
+        verify(mockResult, times(0)).success(
+                mapOf(
+                        "data" to restResponse.data.rawBytes
+                )
+        )
+
+        verify(mockCancelResult).success(
+                "Operation Canceled"
+        )
+    }
+
+    // Cancellation of non existing rest operation fails
+    @Test
+    fun test_cancel_get_returns_error(){
```

Comment:
Can we add another test to verify the `removeOperation`. E.g. make a `get()` -> `cancel()` -> verify success -> `cancel()` -> verify error (operation has already been removed).